File: src/main/java/dominio/indicadores/OperacionAritmetica.java
package dominio.indicadores;

public enum OperacionAritmetica {
	
	Mas{
		public int applyAsInt(int operandoIzq, int operandoDer) {
			return operandoIzq + operandoDer;
		}
	},
	
	Menos{
		public int applyAsInt(int operandoIzq, int operandoDer) {
			return operandoIzq - operandoDer;
		}
	},
	
	Por{
		public int applyAsInt(int operandoIzq, int operandoDer) {
			return operandoIzq * operandoDer;
		}
	},
	
	Dividido{
		public int applyAsInt(int operandoIzq, int operandoDer) {
			return operandoIzq / operandoDer;
		}
	};
	
	public abstract int applyAsInt(int operandoIzq, int operandoDer);

}

File: src/main/java/dominio/indicadores/OperacionAritmetica.kt
package dominio.indicadores

import java.util.function.IntBinaryOperator

enum class OperacionAritmetica: IntBinaryOperator {

    Mas {
        override fun applyAsInt(operandoIzq: Int, operandoDer: Int): Int = operandoIzq + operandoDer
    },

    Menos {
        override fun applyAsInt(operandoIzq: Int, operandoDer: Int): Int = operandoIzq - operandoDer
    },

    Por {
        override fun applyAsInt(operandoIzq: Int, operandoDer: Int): Int = operandoIzq * operandoDer
    },

    Dividido {
        override fun applyAsInt(operandoIzq: Int, operandoDer: Int): Int = operandoIzq / operandoDer
    };

    abstract override fun applyAsInt(operandoIzq: Int, operandoDer: Int): Int

}
